website/pages/docs/developers/creating-new-plugin/java-source.mdx

```diff
@@ -0,0 +1,167 @@
+
+Though you by no means need to be an expert, you will also need some familiarity with Java. If you are new to Java, we recommend starting with the [W3Schools Java tutorials](https://www.w3schools.com/java/).
+
+## Core Concepts
+
+Before we dive in, let's quickly cover some core concepts of CloudQuery plugins, so that they're familiar when we see our first example.
+
+### Syncs
+
+A sync is the process that gets kicked off when a user runs `cloudquery sync`. A sync is responsible for fetching data from a third-party API and inserting it into the destination (database, data lake, stream, etc.). When you write a source plugin for CloudQuery, you will only need to implement the part that interfaces with the third-party API. The rest of the sync process, such as delivering to the destination database, is handled by the CloudQuery SDK.
+
+### Tables and Services
+
+A **table** is the term CloudQuery uses for a collection of related data. In most databases it directly maps to an actual database table, but in some destinations it could be stored as a file, stream or other medium. A table is defined by a name, a list of columns, and a resolver function. The resolver function is responsible for fetching data from the third-party API and sending it to CloudQuery. We will look at examples of this soon!
+
+Every table will typically have its own `.java` file in the plugin `resources` package.
+
+### Resolvers
+
+Resolvers are functions associated with a table that get called when it's time to populate data for that table. There are two types of resolvers:
+
+#### Table resolvers
+
+Table resolvers are responsible for fetching data from the third-party API. In Java, a table resolver is a class that implements the [`TableResolver`](https://github.com/cloudquery/plugin-sdk-java/blob/9c7b78fff2ae72fc2eb0f18e12372bf91bffffd5/lib/src/main/java/io/cloudquery/schema/TableResolver.java) interface.
+
+#### Column resolvers
+
+[Column resolvers](https://github.com/cloudquery/plugin-sdk-java/blob/9c7b78fff2ae72fc2eb0f18e12372bf91bffffd5/lib/src/main/java/io/cloudquery/schema/ColumnResolver.java) are responsible for mapping data from the third-party API into the columns of the table. In most cases, you will not need to implement this, as the SDK will automatically map data from the class passed in by the table resolver to the columns of the table. But in some cases, you may need to implement a custom column resolver to fetch additional data or do custom transformations.
+
+## Apache Arrow Java SDK
+
+Like other CloudQuery SDK the Java SDK uses [Apache Arrow] as part of the underlying CloudQuery type system.
+
+## Creating Your First Plugin
+
+As the Java SDK is still new, we don't yet have a plugin scaffold generator. For now, we recommend starting by copying the code from an [existing Java plugin](https://github.com/cloudquery/cloudquery/tree/main/plugins/source/bitbucket).
+
+<!-- vale off -->
```

Comment:
In `mdx` files this needs to be `{/* <!-- vale off --> */}` (noticed the `{}`), or you can change this file to `md` if you don't use any React components

Reply:
Would this be why Vercel is failing?

Reply:
Yes probably since it breaks MDX syntax